core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/logreplay/ReplayOrderEnforcer.scala

```diff
@@ -0,0 +1,48 @@
+package edu.uci.ics.amber.engine.architecture.logreplay
+
+import edu.uci.ics.amber.engine.common.ambermessage.ChannelID
+
+import scala.collection.mutable
+
+class ReplayOrderEnforcer(
+    logManager: ReplayLogManager,
+    channelStepOrder: mutable.Queue[ProcessingStep],
+    startStep: Long,
+    replayTo: Long,
+    var onComplete: () => Unit
+) extends OrderEnforcer {
+  private var currentChannelID: ChannelID = _
+  var isCompleted: Boolean = startStep > replayTo
+
+  // restore replay progress by dropping some of the entries
+  while (channelStepOrder.nonEmpty && channelStepOrder.head.steps <= startStep) {
```

Comment:
can you change the ProcessinStep.steps  to `step`?

Reply:
done